models/classes/class.QtiTestCompiler.php

```diff
@@ -338,9 +338,23 @@ public function compile() {
             $report->setData($serviceCall);
         }
         catch(XmlStorageException $e){
+
             $subReport = new common_report_Report(common_report_Report::TYPE_ERROR, __('The QTI Test XML or one of its dependencies is malformed or empty.'));
             $report->add($subReport);
 
+            $previous = $e->getPrevious();
+            while (true) {
+
+                if (!$previous instanceof Exception) {
+                    break;
+                }
+
+                $subReport = new common_report_Report(common_report_Report::TYPE_ERROR, __($previous->getMessage()));
```

Comment:
This will be hard to translate as we do not know the nature of the various possible messages.